test/testCore.cpp

```diff
@@ -435,15 +435,36 @@ TEST(ScalarGrid, SimpleTest) {
 	EXPECT_FLOAT_EQ(7., grid.interpolate(some_grid_point));
 }
 
-TEST(ScalarGrid, TestVectorSpacing) {
+TEST(Grid1f, GridPropertiesConstructor) {
 	// Test constructor for vector spacing
 	size_t Nx = 5;
 	size_t Ny = 8;
 	size_t Nz = 10;
 	Vector3d origin = Vector3d(1., 2., 3.);
 	Vector3d spacing = Vector3d(1., 5., 3.);
+	GridProperties properties(origin, Nx, Ny, Nz, spacing);
+	Grid1f grid(properties);
 
-	ScalarGrid grid(origin, Nx, Ny, Nz, spacing);
+	EXPECT_EQ(spacing, grid.getSpacing());
+
+	// Test index handling: get position of grid point (1, 7, 6)
+	size_t some_index = 1 * Ny * Nz + 7 * Nz + 6;
+	Vector3d some_grid_point = origin + Vector3d(1, 7, 6) * spacing + spacing / 2.;
+
+	grid.get(1, 7, 6) = 12;
```

Comment:
The Grid1f.grid function acts also as a setter/mutator right?

Looks a bit unfamiliar but if it works it's probably fine here.

Reply:
We could consider renaming the function to .index()? Maybe this would make it clearer that it's not just a getter. I don't think it's that important though either.

Reply:
I don't a strong opinion on that. We could follow your suggestion or introduce a new function .set() and remove the setting capabilities from .get(), if that is possible.

Reply:
I'll leave this for later PR since it affects the standard interface I think.

Reply:
Fine for me.